src/Sarif.Multitool/Rules/SarifValidationSkimmerBase.cs

```diff
@@ -554,6 +662,36 @@ private void Visit(Region region, string regionPointer)
             Analyze(region, regionPointer);
         }
 
+        private void Visit(ReportingConfiguration reportingConfiguration, string reportingConfigurationPointer)
+        {
+            Analyze(reportingConfiguration, reportingConfigurationPointer);
+        }
+
+        private void Visit(ReportingDescriptor reportingDescriptor, string reportingDescriptorPointer)
+        {
+            Analyze(reportingDescriptor, reportingDescriptorPointer);
+
+            if (reportingDescriptor.ShortDescription != null)
+            {
+                Visit(reportingDescriptor.ShortDescription, reportingDescriptorPointer.AtProperty(SarifPropertyName.ShortDescription));
+            }
+
+            if (reportingDescriptor.FullDescription != null)
+            {
+                Visit(reportingDescriptor.FullDescription, reportingDescriptorPointer.AtProperty(SarifPropertyName.FullDescription));
+            }
+        }
+
+        private void Visit(ReportingDescriptorReference reportingDescriptorReference, string reportingDescriptorReferencePointer)
```

Comment:
>Visit [](start = 21, length = 5)

this code never should have reauthored a comprehensive visitor in the SDK but should have used our rewriting visitor. all this churn emphasizes the point to me.

Reply:
I don't remember how we got to this state. I think I discovered at some point that the generated `RewritingVisitor` wasn't fit for this purpose, so I started work in a branch (which I still have hanging around called) `users/lgolding/generate-visitor`, where I was going to generate _this_ visitor. But I never got time to finish it, and it turns out there are aspects of this code that are hard to auto-generate. I'm not opposed to taking another look, but boy, I can't imagine this rising to the top of my stack.

---
In reply to: [305431532](https://github.com/microsoft/sarif-sdk/pull/1570#discussion_r305431532) [](ancestors = 305431532)

Reply:
There are places in this code where I have to set context to establish whether I'm looking at a notification descriptor, or a taxon descriptor, in order to produce the correct error message. (See, for example, the method `Visit(Notification, string)` in this class, and in SARIF1017.InvalidIndex.cs, see the overload of `Analyze` that takes a `ReportingDescriptorReference`.)

That would make it hard to auto-generate this class.

---
In reply to: [305526382](https://github.com/microsoft/sarif-sdk/pull/1570#discussion_r305526382) [](ancestors = 305526382,305431532)